[DRAFT] Consolidate locations logic in order to support Beta updates

coffeemancy opened the pull request:
The intention of this PR is some cleanup preempting changes to add support for Version 3.3 (current Beta) features of the randomizer in follow-on PRs, e.g. extra KI/spots, Epoch Fail / Unlocked Skyways changes, and Rocksanity. Additionally, support for Ice Age mode (which is currently missing) could be added in a smaller, follow-on PR.

It also means that when future updates are made, only a single locations file will need updating. Currently there is a lot of repetition through many copy-pasted files, which is error prone, as well as painful to maintain. (For example, in doing this refactoring, I found the Sun Keep spot sometimes mentioned "Sunstone" instead of "sunstone" and was not showing up as a location on the map tracker).

## Updates

NOTE: This builds off of https://github.com/Aeralis/Jets-of-Time-Tracker/pull/5 which is expected to be merged first (covering up to commit https://github.com/Aeralis/Jets-of-Time-Tracker/commit/cb70d484bf0f64167f73bfa883a8c8743398d847). Until that is merged, it might be a bit difficult to review this PR entirely.

The main additions this PR makes (starting at https://github.com/Aeralis/Jets-of-Time-Tracker/commit/14512e78bf3d328967c572f144953ea8c675bbd5):

* Removes seemingly unused "chronosanity" files
* Removes separate locations files for "legacy_of_cyrus", "lost_worlds", and "vanilla_rando"
* Consolidates all locations logic in `locations/locations.json`
* Updates `scripts/` to include mode-based logic (used in locations for access/visibility rules depending on e.g. Legacy of Cyrus, Lost Worlds, Vanilla Rando, etc.)

## Testing

The only testing I've done so far is using PopTracker and viewing the separate modes and manually checking that it seems to work as expected. With each consolidation, I checked that the spots on the maps looked the same (although not exhaustively).

All files pass PopTracker JSON schema validation and no warnings are output in trying the different modes.

Additional testing definitely warranted. :crossed_fingers: this works on EmoTracker...

coffeemancy commented on scripts/logic.lua:
```diff
@@ -30,6 +34,67 @@ function canAccessSealed()
+  if legacyOfCyrusMode() then
+    local frog = Tracker:FindObjectForCode("frog").Active
+    local magus = Tracker:FindObjectForCode("magus").Active
+    local magusboss = Trakcer:FindObjectForCode("magusboss").Active
```
Typo "Trakcer" -> "Tracker".

Commit:
rm unused chronosanity cruft

Commit:
move mode-checking to shared logic.lua file

Commit:
remove empty access_rules which PopTracker warns about

Commit:
consolidate std/LoC locations, mode-based access/visibility rules

Commit:
also consolidate Lost Worlds locations

Commit:
consolidate Vanilla Rando mode

Commit:
bump version, changelog

Anguirel86 commented on chronosanity/layouts/broadcast.json:
Good catch on these.  Chronosanity was consolidated with the other variants as a flag in the last major update.  I guess I forgot to git rm the old files, so they stuck around.

Commit:
Fix tracking for Dupe Characters; use Github Releases; lua/JSON checking

Squashed commit of "dupe characters" changes.

commit 05a5643c54009173971243a2293580efb9eb133b
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Mon Aug 21 21:26:27 2023 -0400

    revert items.json changes, EmoTracker does use "loop" (#8)

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit cb70d484bf0f64167f73bfa883a8c8743398d847
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Mon Aug 21 20:01:52 2023 -0400

    PopTracker JSON schema validation (#6)

    * implement JSON schema validation using PopTracker schemas

    * update items.json to pass strict JSON schema validation

    * separate "pip" step in workflow, verbose

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit 2021a51723948559b7f13d744d5f7e7ea7e1a0b8
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Sat Aug 5 21:10:36 2023 -0400

    Use memory watch to handle autotracking equipable items (#5)

    * fix typo, point to upstream repo for Releases link in README

    * use memory watch to handle equipable items

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit e9a4631488aa0db988ebb43dd22d2c42750df085
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Fri Aug 4 21:40:51 2023 -0400

    Fix handling equippable items to work with Duplicate Characters (#4)

    * fix CONTRIBUTING typo

    * fix handling equipable items to work with duplicate characters

    Previously encountered bug that when duplicate characters is enabled,
    if a frog duplicate equips Hero Medal or Grand Leon, and is not the "Frog"
    character (e.g. a "Marle" who is a frog duplicate), then tracker
    will not show the item anymore. This fixes that by scanning all
    characters for the item, to work with duplicate characters flag.

    * update changelog

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit 161b4da3860c504297c334ade95840f32ea8e713
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Fri Aug 4 21:24:33 2023 -0400

    Move to using Github releases (#3)

    * rm .zip files, use github Releases

    * add README and CONTRIBUTING files

    * more consistent naming, bump to 2.1.1

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit 8ed4b68d870f2276a68dd6162ed0efe11dd9f9e3
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Fri Aug 4 20:08:17 2023 -0400

    Add strict JSON test to github workflow and get JSON files to pass (#2)

    * add JSON tester to run on PR/merge

    * rm invalid JSON (comments, trailing commas)

    Comments and trailing commas (in a data structure's last element)
    are invalid JSON. Removed them.

    * don't need to run tests twice, just on push

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

commit d52cd5b6d9cd8ecfd03b58701f9ef84c7d95e9e3
Author: C <3493534+coffeemancy@users.noreply.github.com>
Date:   Fri Aug 4 16:16:55 2023 -0400

    Luacheck and whitespace cleanup (#1)

    * consistent EOL characters for windows/mac/linux devs

    Set .gitattributes to make sure line endings are converted on
    checkout based on platform (i.e. CR+LF for Windows, LF for
    Mac/Linux).

    * github workflow to run luacheck on changes to .lua files

    * do not add .git* files or *.zip during `git archive`

    * tabs->spaces and remove extraneous whitespace from lua files

    * run luacheck on workflow changes, ignore globals for now

    * set higher max line length, as code currently meets

    * pass luacheck

    ---------

    Co-authored-by: C <caltronnotlrac@gmail.com>

coffeemancy commented:
Replaced with https://github.com/Aeralis/Jets-of-Time-Tracker/pull/7
